### DISCUSSION TITLE
Access Media Azure files in MVC application without CMS

### DISCUSSION THREAD

**Discussion by vsezima** [OrchardCMS/OrchardCore#14660] [General]
### Hello, I had question on Discussions page, but no reply, so I'm trying again on Issues page, hoping more people will see it.
Discussed in https://github.com/OrchardCMS/OrchardCore/discussions/14582

<div type='discussions-op-text'>

<sup>Originally posted by **vsezima** October 26, 2023</sup>
Hello,
I'm having trouble accessing Azure media files in MVC application without CMS (created with this tutorial https://docs.orchardcore.net/en/latest/docs/guides/create-modular-application-mvc/) and was wondering, if it is even possible.

When I create CMS website (with this tutorial https://docs.orchardcore.net/en/latest/docs/guides/create-cms-application/) and add Azure media feature and proper appsettings configuration, then I see Azure files in CMS media library on URL /admin/media and can access them directly on URL /media/filename.

For MVC application, I added needed references and appsettings, but cannot access same file on URL /media/filename (it returns 404).
.csproj references:
![image](https://github.com/OrchardCMS/OrchardCore/assets/15254338/c8161332-6466-48fb-bcad-55c0d54a8821)
Appsettings configuration:
![image](https://github.com/OrchardCMS/OrchardCore/assets/15254338/0b920aa4-532c-4929-ac2e-e80897fb639d)
Program.cs:
![image](https://github.com/OrchardCMS/OrchardCore/assets/15254338/e339e05a-ce6b-4885-9e18-547953f0a9a0)

In Program.cs I had to add 3 extra extensions methods so URL /media/filename wouldn't throw exceptions about missing services, so I think that Media feature works, but it still can't find requested file.</div>

**Comment by sebastienros** [comment]
The `/media/filename` is provided by the Media feature which probably requires a bunch of other ones (including CMS ones).

**Reply by vsezima** [reply]
Yes, we discussed it further here https://github.com/OrchardCMS/OrchardCore/discussions/14582 and I dug through code to find many dependencies on CMS.
I guess it's too much work to separate the dependencies, so I will try to make my own endpoint and fileprovider for Azure blob storage.
